flink-kubernetes/src/main/java/org/apache/flink/kubernetes/cli/KubernetesSessionCli.java

```diff
@@ -88,6 +88,7 @@ public Configuration getEffectiveConfiguration(String[] args) throws CliArgsExce
 
 	private int run(String[] args) throws FlinkException, CliArgsException {
 		final Configuration configuration = getEffectiveConfiguration(args);
+		KubernetesClusterClientFactory.ensureClusterIdIsSet(configuration);
```

Comment:
Shouldn't we obtain the `clusterId` from the `KubernetesClusterDescriptor`?

Reply:
That way we wouldn't have this clumsy call here.

Reply:
The `(Kubernetes)ClusterDescriptor` provides no way of accessing the cluster-id.
There is `ClusterClientFactory#getClusterId`, but this one only works the way we'd like to after `ClusterClientFactory#createClusterDescriptor` was called (since it modifies the passed configuration).

I thought about changing `ClusterClientFactory#getClusterId` to also set the id if it is missing, to ensure that the behavior of the methods is identical regardless of method call order, but I cannot assess what repercussions this might have on users of the `ClusterClientFactory` interface.

Basically, I hate this behavior, and didn't want to rely on it.
I would argue that the user of the `ClusterClientFactory` should ensure that the cluster-id was set, but I didn't want to remove the auto-generation in `ClusterClientFactory#createClusterDescriptor` because god knows what it might break.

Reply:
Hmm, yeah one should either get the `clusterId` from the user or after a cluster is deployed via the `ClusterClient` or `ClusterClientProvider`.